test/e2e/scenario/recorder/shadowDom.scenario.ts

```diff
@@ -238,3 +295,7 @@ async function getNodeInsideShadowDom(hostTag: string, selector: string) {
   const host = await $(hostTag)
   return host.shadow$(selector)
 }
+
+function isAdoptedStyleSheetsSupported() {
+  return ['edge', 'chrome'].includes(getBrowserName())
+}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Could we check the `adoptedStyleSheets` property on document instead? 
it would allow to automatically run this test on other browsers when available

Reply:
no because it would be in the runner (node) not in the targetted browser :(

Reply:
Maybe indeed, @BenoitZugmeyer you suggest that approach. Maybe you want to share your opinion here

Reply:
Is it possible to get the result out of the browser to the runner to stop the test?

Reply:
It's a bit more annoying as it would be asynchronous, but we can try!

Note, instead of `if (...) { createTest(...).run() }` it would need to done in the `.run()` like:
```ts
.run(async () => {
  if (!await isAdoptedStyleSheetsSupported()) { pending('...') }
  ...
})
```

Reply:
@BenoitZugmeyer How can I test that locally? By default it will run on Chrome up to date and be ok with the test but how can I test that it will be pending?

Reply:
You can try with a constant condition (ex: `if (!await Promise.resolve(false)) { pending() }`

About the condition specifically, I managed to run e2e tests in firefox locally, or you can run your tests on browserstack with the `yarn test:e2e:bs` with the right tokens as environment variable. Or simply push the change and see how it goes :)